plugins/modules/aci_interface_config.py

```diff
@@ -0,0 +1,439 @@
+    node = module.params.get("node")
+    pc_member = module.params.get("pc_member")
+    port_type = module.params.get("port_type")
+    role = module.params.get("role")
+    admin_state = module.params.get("admin_state")
+    interface = module.params.get("interface")
+    interface_type = module.params.get("interface_type")
+    state = module.params.get("state")
+
+    aci = ACIModule(module)
+
+    if node is not None and node not in range(101, 4001):
+        aci.fail_json(msg="Node ID: {0} is invalid; it must be in the range of 101 to 4000.".format(node))
+
+    card, port_id, sub_port = (None, None, None)
+    if interface is not None:
+        interface_parts = interface.split("/")
+        if len(interface_parts) == 3:
+            card, port_id, sub_port = interface_parts
+        elif len(interface_parts) == 2:
+            card, port_id = interface_parts
+            sub_port = 0
+        else:
+            aci.fail_json(msg="Interface: {0} is invalid; The format must be either card/port/sub_port(1/1/1) or card/port(1/1)".format(interface))
+
+        if card is not None and int(card) not in range(1, 65):
+            aci.fail_json(msg="Card ID: {0} is invalid; it must be in the range of 1 to 64.".format(card))
+
+        if port_id is not None and int(port_id) not in range(1, 129):
+            aci.fail_json(msg="Port ID: {0} is invalid; it must be in the range of 1 to 128.".format(port_id))
+
+        # Sub Port ID - 0 is default value
+        if sub_port is not None and int(sub_port) not in range(0, 17):
+            aci.fail_json(msg="Sub Port ID: {0} is invalid; it must be in the range of 0 to 16.".format(sub_port))
+
+    root_class = PORT_TYPE_MAPPING.get(port_type)["root_class"]
+    root_class_rn = PORT_TYPE_MAPPING.get(port_type)["root_class_rn"]
+    interface_class = PORT_TYPE_MAPPING.get(port_type)["interface_class"]
+
+    aci.construct_url(
+        root_class=dict(
+            aci_class=root_class,
+            aci_rn=root_class_rn,
+        ),
+        subclass_1=dict(
+            aci_class=interface_class,
+            aci_rn="portconfnode-{0}-card-{1}-port-{2}-sub-{3}".format(node, card, port_id, sub_port),
+            target_filter=dict(node=node),
+        ),
+    )
+
+    aci.get_existing()
+
+    if breakout is None and policy_group:
+        policy_group_dn = POLICY_GROUP_MAPPING.get(interface_type).format(policy_group)
+    else:
+        # To handle the existing object property
+        policy_group_dn = ""
+
+    if state == "present":
+        aci.payload(
+            aci_class=interface_class,
+            class_config=dict(
+                assocGrp=policy_group_dn,
+                brkoutMap=breakout,
+                card=card,
+                description=description,
+                node=node,
+                pcMember=pc_member,
+                port=port_id,
+                role=role,
+                shutdown=ADMIN_STATE_MAPPING.get(admin_state, ""),
```

Comment:
admin_state has default from input so will it ever be None? Or should we not use the defaults during input?